src/components/table/table-modern.style.js

```diff
@@ -45,27 +47,49 @@ function applyModernTableStyling({
       `}
     }
     ${type !== 'primary' && additionalThemeStyling(type, theme)}
+    ${addInputStyling(size)}
   `;
 }
 
 function additionalThemeStyling(type, { text, table, colors }) {
   return css`
-    ${type === 'tertiary' ? 'border-color: transparent;' : ''}
-    ${StyledTableCell} {
-      background-color: ${colors.white};
-    }
-
-    ${StyledTableHeader} {
-      background-color: ${type === 'secondary' ? table.secondary : 'transparent'};
-      ${type === 'tertiary' ? 'border-left-color: transparent;' : ''}
-      color: ${text.color};
-        
-      a:link,
-      a:visited,
-      a:hover,
-      a:active {
+      ${type === 'tertiary' ? 'border-color: transparent;' : ''}
+      ${StyledTableCell} {
+        background-color: ${colors.white};
+      }
+  
+      ${StyledTableHeader} {
+        background-color: ${type === 'secondary' ? table.secondary : 'transparent'};
+        ${type === 'tertiary' ? 'border-left-color: transparent;' : ''}
         color: ${text.color};
+          
+        a:link,
+        a:visited,
+        a:hover,
+        a:active {
+          color: ${text.color};
+        }
       }
+    `;
+}
+
+function addInputStyling(size) {
+  return `
+    ${StyledInput} {
+      font-size: ${tableSizes[size].font};
+      height: ${tableSizes[size].inputHeight};
+      padding-left: ${tableSizes[size].padding};
+      padding-right: ${tableSizes[size].padding};
+    }
+
+    && ${StyledInputPresentation} {
+      height: ${tableSizes[size].inputHeight};
+      min-height: ${tableSizes[size].inputHeight};
+    }
+
+    && ${StyledTableCell} div {
+      height: ${tableSizes[size].inputHeight};
+      min-height: ${tableSizes[size].inputHeight};
```

Comment:
it could be simplified to:
```
const { font, inputHeight, padding } = tableSizes[size];
```
I would also consider renaming `font: fontSize` and `padding: paddingSize` during destructuration

Reply:
Yeah good call, I've made the changes